src/core/ext/xds/xds_api.cc

```diff
@@ -399,6 +402,23 @@ absl::Status XdsApi::ParseAdsResponse(const XdsBootstrap::XdsServer& server,
         "type.googleapis.com/");
     absl::string_view serialized_resource =
         UpbStringToAbsl(google_protobuf_Any_value(resources[i]));
+    // Unwrap Resource messages, if so wrapped.
+    if (type_url == "envoy.service.discovery.v2.Resource" ||
```

Comment:
v2 should be `envoy.api.v2.Resource`, right?

Reply:
Good catch!  Fixed.

(I don't expect that we'll actually ever use this in v2; I added it here just for completeness.)